tests/test_loaders.py

```diff
@@ -40,4 +42,47 @@ def test_load_sound_speed(all_files_dict):
     svdf = load_sound_speed(all_files_dict["sound_speed"])
     assert isinstance(svdf, DataFrame)
     assert {SP_DEPTH, SP_SOUND_SPEED} == set(svdf.columns.values.tolist())
-    assert svdf.dtypes[SP_DEPTH] == float64 and svdf.dtypes[SP_SOUND_SPEED] == float64
+    assert is_float_dtype(svdf[SP_DEPTH]) and is_float_dtype(svdf[SP_SOUND_SPEED])
+
+
+@pytest.mark.parametrize(
+    "time_round",
+    [3, 6],
+)
+def test_load_gps_solutions(all_files_dict, time_round):
+    loaded_gps_solutions = load_gps_solutions(
+        all_files_dict["gps_solution"], time_round
+    )
+    expected_columns = [GPS_TIME, *GPS_GEOCENTRIC, *GPS_COV]
+
+    assert isinstance(loaded_gps_solutions, DataFrame)
+    assert set(expected_columns) == set(loaded_gps_solutions.columns.values.tolist())
+    assert all(
+        is_float_dtype(loaded_gps_solutions[column])
+        for column in loaded_gps_solutions.columns
+    )
+
+    raw_gps_solutions = pd.concat(
+        [
+            read_csv(i, delim_whitespace=True, header=None, names=expected_columns)
+            for i in all_files_dict["gps_solution"]
+        ]
+    ).reset_index(drop=True)
+
+    # Dimension of raw_gps_solutions df should equal loaded_gps_solutions df
+    assert loaded_gps_solutions.shape == raw_gps_solutions.shape
+
+    # Verify rounding decimal precision of GPS_TIME column
+    gps_times = pd.concat(
+        [loaded_gps_solutions[GPS_TIME], raw_gps_solutions[GPS_TIME]],
+        axis=1,
+        keys=[f"loaded_gps_solutions_{GPS_TIME}", f"raw_gps_solutions_{GPS_TIME}"],
+    )
+    gps_times["equality"] = gps_times.apply(
+        lambda row: isclose(
+            row[f"loaded_gps_solutions_{GPS_TIME}"],
+            row[f"raw_gps_solutions_{GPS_TIME}"].round(time_round),
+        ),
+        axis=1,
+    )
+    assert gps_times["equality"].all()

```

Comment:
This is great! In addition, could you also verify that the values are rounded by the specified `time_round`? You should be able to get the number of digits after the decimal points.

Reply:
I am already verifying the rounding of `GPS_TIME` column by `time_round` precision in line 81-88.

Reply:
Ah gotcha. Sorry I missed that. Could this be compared with `equal` rather than `isclose`? What is the reason behind `isclose`?

Reply:
I was wrongly under the impression that numpy float rounding would lead to some floating point imprecisions (at the last few significant figures); so I used `isclose` rather than `equals`. I have created a PR and reverted to using `equals`